test/unit/httpd_role_test.lua

```diff
@@ -268,3 +269,32 @@ g['test_get_server_bad_type'] = function()
     t.assert_not(ok)
     t.assert_str_contains(res, '?string expected, got table')
 end
+
+g['test_log_requests_default'] = function()
+    local cfg = {
+        [httpd_role.DEFAULT_SERVER_NAME] = {
+            listen = 13002,
+        },
+    }
+
+    httpd_role.apply(cfg)
+
+    local result = httpd_role.get_server()
+    t.assert_not_equals(result, nil)
+    t.assert_equals(result.options.log_requests, false)
+end
+
+g['test_log_requests_custom'] = function()
+    local cfg = {
+        [httpd_role.DEFAULT_SERVER_NAME] = {
+            listen = 13003,
+            log_requests = log.verbose,
+        },
+    }
+
+    httpd_role.apply(cfg)
+
+    local result = httpd_role.get_server()
+    t.assert_not_equals(result, nil)
+    t.assert_equals(result.options.log_requests, log.verbose, "Should use custom log level")
+end
```

Comment:
I think it can be merged into one test:

* create config with 2 servers with other `log_requests` options;
* check it in for loop (for example). Something like this:

```lua
    for name, body in pairs(cfg) do
        local res = httpd_role.get_server(name)
        t.assert(res)
        t.assert_equals(res.options.log_requests, body.log_requests or false)
    end
```